src/main/java/mate/academy/shop/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,191 @@
+                String surname = resultSet.getString(USER_SURNAME_CONST);
+                String login = resultSet.getString(USER_LOGIN_CONST);
+                String password = resultSet.getString(USER_PASSWORD_CONST);
+                String token = resultSet.getString(USER_TOKEN_CONST);
+                User user = new User(userId);
+                user.setName(name);
+                user.setSurname(surname);
+                user.setLogin(login);
+                user.setPassword(password);
+                user.setToken(token);
+                list.add(user);
+            }
+        } catch (SQLException e) {
+            logger.warn("Can't get users");
+        }
+        return list;
+    }
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        String query = "SELECT * FROM users WHERE login=? AND password=?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            statement.setString(2, password);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                long userId = resultSet.getLong(USER_ID_CONST);
+                String name = resultSet.getString(USER_NAME_CONST);
+                String surname = resultSet.getString(USER_SURNAME_CONST);
+                String token = resultSet.getString(USER_TOKEN_CONST);
+                User user = new User(userId);
+                user.setName(name);
+                user.setSurname(surname);
+                user.setLogin(login);
+                user.setPassword(password);
+                user.setToken(token);
+                return user;
+            }
+        } catch (SQLException e) {
+            logger.warn("Incorrect username or password");
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<User> getByToken(String token) {
+        String query = "SELECT * FROM users WHERE token=?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, token);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long userId = resultSet.getLong(USER_ID_CONST);
+                String name = resultSet.getString(USER_NAME_CONST);
+                String surname = resultSet.getString(USER_SURNAME_CONST);
+                String login = resultSet.getString(USER_LOGIN_CONST);
+                String password = resultSet.getString(USER_PASSWORD_CONST);
+                String tokenUsers = resultSet.getString(USER_TOKEN_CONST);
+                User user = new User(userId);
+                user.setName(name);
+                user.setSurname(surname);
+                user.setLogin(login);
+                user.setPassword(password);
+                user.setToken(tokenUsers);
+                return Optional.of(user);
```

Comment:
@TarasHryts Can you move it to a separate method?

Reply:
I can, but I do not think that create the method with 5 or more variables is a good idea

Reply:
if you create this method, you can remove repeatable code in some methods